### ISSUE TITLE
ESC key closes entire sub-menus all the way

### ISSUE THREAD

**Issue by jooyoungseo** [rstudio/rstudio#6327]
### System details

RStudio Edition : Server
RStudio Version : 1.3.866
OS Version      : Windows 10
R Version       : any

### Steps to reproduce the problem

To clearly understand this request, please examine keyboard menu access behavior in Windows default application.

1. Run "notepad" on any Windows systems.
2. Press Alt+V followed by Z, then your system cursor will land in the following menus:

> "View" (pull-down menu) > "Zoom" (sub menu) > "Zoom In"

3. Pres ESC key to see what happens: ESC key closes collapsible menu each time pressed instead of closing the entire opened menus at once.

### Describe the problem in detail

ESC key cloeses all the sub-menus opened. Due to this behavior, keyboard navigation for menus in RStudio Server widget is not as convenient as Windows default application's for those who cannot use mouse.

### Describe the behavior you expected

It would be ideal ESC key behaves as in Windows default applications like notepad.

**Comment by gtritchie** [comment]
As a workaround you can use the left-arrow key to close a submenu without closing the entire menu.

This is consistent with macOS menus, actually, but not Windows.

The ARIA practices for menus does say that ESC should close the submenu and move focus to the parent menu, as you describe.

https://www.w3.org/TR/wai-aria-practices/#menu

**Comment by jooyoungseo** [comment]
Thanks, @gtritchie! I know this could be a workaround and convension on MacOS. However, users will have to press ESC key (not left-arrow key) to get out of the main menu bar at the end, which is not a consistent navigation mode I think.

**Comment by gtritchie** [comment]
Noticed a related case; reopening to handle that as part of this issue.

## Steps
Bring up a Server main menu item with a keyboard shortcut, say <kbd>Ctrl+Opt+F</kbd> on Mac or <kbd>Alt+Shift+F</kbd> on Windows/Linux for the File menu.

Focus goes to the main menubar (the horizontal menu containing File, Edit, etc.) and expands the menu beneath the selected item (File in this case), leaving focus on the File menu item itself.

Without doing anything else, hit ESC.

## Problem
The file popup menu closes, the File selection disappears, but keyboard focus actually remains on the menubar; if you then hit right-arrow, it will select File.

## Expected
Hitting ESC when focus is on the main menubar itself, but the current selection is expanded, should just close the popup and leave focus on the current item.

**Comment by kfeinauer** [comment]
@gtritchie Looks like menu focus is a bit wonky for my Windows test environment.

I am running Open Source server latest master on my OSX machine, and connecting my Windows machine to it via browser (tested both Chrome and Edge).

Whenever I see hit `Alt+Shift+F` and traverse a menu tree, whenever I hit escape, all menus close. Then, focus seems to be lost - it is not returned to where it was previously (in my case, the Console). Hitting left and right arrow keys at this point does nothing.

**Comment by kfeinauer** [comment]
This was a false alarm - verified this is working with RSP 1.4.953-1
